portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,35 @@
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+   
+@WebServlet("/delete-data") 
+public class DeleteDataServlet extends HttpServlet {
+  private String commentEntityName = "Comment";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query(commentEntityName);
+    PreparedQuery allComments = datastore.prepare(query);
+    for(Entity entity : allComments.asIterable()){
+      long id = entity.getKey().getId();
+      Key commentEntityKey = KeyFactory.createKey(commentEntityName, id);
```

Comment:
Don't need to make a new key since entity already has the key.

Reply:
Good catch! I only wrote it this way since I was emulating the walkthrough. I wonder why they had made it like that in the first place.